oci/platform.py

```diff
@@ -160,3 +211,57 @@ def iter_platforms(
         )
 
         yield (sub_img_ref, platform)
+
+
+class PlatformFilter:
+    @staticmethod
+    def create(
+        included_platforms: typing.List[str],
+    ) -> typing.Callable[[om.OciPlatform], bool]:
+        matchers = []
+        for included_platform in included_platforms:
+            matchers.append(PlatformFilter._parse_expr(included_platform))
+
+        def filter(platform_to_match: om.OciPlatform) -> bool:
+            for m in matchers:
+                if PlatformFilter._match(m, platform_to_match):
+                    return True
+
+            return False
+
+        return filter
+
+    @staticmethod
+    def _parse_expr(platform_expr: str) -> dict:
+        splitted = platform_expr.split('/')
+        if len(splitted) < 2 or len(splitted) > 3:
```

Comment:
rather use separate if-clauses for both variants - or at least output actual length in error-msg to make debugging easier

Reply:
you might want to consider using structural pattern matching (https://peps.python.org/pep-0622/)